forest/daemon/src/main.rs

```diff
@@ -139,93 +140,73 @@ type Db = forest_db::rocks::RocksDb;
-                                cli_error_and_die(
-                                    format!("Error when detaching. Error was: {e}"),
-                                    1,
-                                );
-                            }
-                        }
-                    }
-
-                    let rt = Runtime::new().unwrap();
-                    if opts.tokio_console {
-                        console_subscriber::init();
-                    }
-
-                    let db: Db = rt.block_on(daemon::start(cfg, opts.detach));
-
-                    info!("Shutting down tokio...");
-                    rt.shutdown_timeout(Duration::from_secs(10));
-
-                    if let Err(e) = db.flush() {
-                        error!("Error flushing db: {e}");
-                    }
-                    let db_weak_ref = Arc::downgrade(&db.db);
-                    drop(db);
-
-                    if db_weak_ref.strong_count() != 0 {
-                        error!(
-                            "Dangling reference to DB detected: {}. Tracking issue: https://github.com/ChainSafe/forest/issues/1891",
-                            db_weak_ref.strong_count()
-                        );
-                    }
-                    info!("Forest finish shutdown");
-                }
+            ConfigPath::Project(path) => {
+                info!("Project config loaded: {}", path.display())
             }
+            _ => (),
         }
-        Err(e) => {
-            logger::setup_logger(&LogConfig::default(), opts.color.into());
-            cli_error_and_die(format!("Error parsing config: {e}"), 1);
+        check_for_unknown_keys(path.to_path_buf(), &cfg);
+    } else {
+        info!("Using default {} config", cfg.chain.name);
+    }
+    match cmd {
+        Some(_) => {
+            warn!("All subcommands have been moved to forest-cli tool");
+        }
+        None => {
+            check_for_low_fd(&cfg).context("Error checking low fd")?;
+            if opts.detach {
+                create_ipc_lock();
+                info!(
+                    "Redirecting stdout and stderr to files {} and {}.",
+                    cfg.daemon.stdout.display(),
+                    cfg.daemon.stderr.display()
+                );
+                let result = build_daemon(&cfg.daemon)
+                    .context("Error building daemon.")?
+                    .start();
+                result.context("Error when detaching.")?;
+                info!("Process detached");
+            }
+
+            let rt = Runtime::new()?;
+            if opts.tokio_console {
+                console_subscriber::init();
+            }
+
+            let db: Db = rt.block_on(daemon::start(cfg, opts.detach))?;
+
+            info!("Shutting down tokio...");
+            rt.shutdown_timeout(Duration::from_secs(10));
+
+            if let Err(e) = db.flush() {
+                error!("Error flushing db: {e}");
+            }
```

Comment:
Let's promote this to a fatal error.

```suggestion
            db.flush()?;
```

Reply:
Fixed